### ISSUE TITLE
[rewrite] Only the last top-level Mithril mountpoint is redrawn implicitly

### ISSUE THREAD

**Issue by barneycarroll** [lhorie/mithril.js#1146]
### Description:

When multiple mountpoints are used, only the last one will be redrawn after Mithril-bound event handlers resolve

### Steps to Reproduce:

[jsbin here](http://jsbin.com/teqedef/edit?js,output)

```js
var Clock  = {
	view : function( component ){
  	return [
      m( 'p', 
        'The time is: ', 
        Date.now() ),

      m( 'button', {
        onclick   : m.redraw
      }, 'Explicit redraw' ),
      
      m( 'button', {
        onclick   : function(){}
      }, 'Implicit redraw' )
    ]
  }
}

document.write( 
    '<div id="site1"></div>'
  + '<div id="site2"></div>' 
)

m.mount( site1, Clock )

m.mount( site2, Clock )
```

### Expected:

After an event handler defined in virtual DOM resolves, all mounted components redraw

### Actual:

Only explicit calls to `m.redraw` redraw globally. Only the last component to be mounted will redraw as a result of event handler resolution.

**Comment by tivac** [comment]
I never intended `m.mount` to support multiple mount-points, so this isn't terribly surprising.

**Comment by barneycarroll** [comment]
What surprises me is that the call graph for auto redraw is so convoluted — and divergent from a call to `m.redraw` (which works as expected).

To my mind, they are the same thing. I tried following the call graph from the event handler wrapper, but after X layers of indirection I thought I'd be better off looking into the auto redraw mechanism itself. Without comments it's difficult to hazard a guess as to what it's intent is and why it's written the way it is, but my inference is that the high level of indirection is there to allow for dependency injection — for the sake of tests, or a potential future path to swapping renderers. But that doesn't explain why manual calls to redraw simply do the intended thing, which AFAICT is intended to be identical in effect — and works.

**Comment by tivac** [comment]
It's changed a lot since it was first introduced in the rewrite.

**Comment by barneycarroll** [comment]
Fixed by #1153
